packages/awssociate/lib/factory.ts

```diff
@@ -84,6 +85,19 @@ export class ServiceFactory<T, U> implements IWithOptions<T, U>, IWithRoleChain<
     }
 }
 
+/**
+ * Adds AWS XRay decoration to the constructed XRay client, if it has been specified as a dependency.
+ */
+function decorateWithXRayIfAvailable<T>(service: T): T{
+    try{
+        const AWSXRay = require('aws-xray-sdk')
```

Comment:
you probably want to use the dynamic import eg;

```
const xray = await import('aws-xray-sdk')
```
As that will work in browser environments as well as via node. Not that I'm entirely sure what kind of fruit-loop would be doing role-assumption chaining on the client side...